lite2/verifier.go

```diff
@@ -31,34 +40,34 @@ func Verify(
 	}
 
 	// Ensure last header can still be trusted.
-	expirationTime := h1.Time.Add(trustingPeriod)
-	if !expirationTime.After(now) {
-		return ErrOldHeaderExpired{expirationTime, now}
+	if HeaderExpired(h1, trustingPeriod, now) {
+		return ErrOldHeaderExpired{h1.Time.Add(trustingPeriod), now}
 	}
 
 	if err := verifyNewHeaderAndVals(chainID, h2, h2Vals, h1, now); err != nil {
-		return err
+		return ErrInvalidNewHeader{err}
 	}
 
 	if h2.Height == h1.Height+1 {
 		if !bytes.Equal(h2.ValidatorsHash, h1NextVals.Hash()) {
-			return errors.Errorf("expected old header validators (%X) to match those from new header (%X)",
+			err := errors.Errorf("expected old header next validators (%X) to match those from new header (%X)",
 				h1NextVals.Hash(),
 				h2.ValidatorsHash,
 			)
+			return ErrNewValSetCantBeTrusted{err}
 		}
 	} else {
 		// Ensure that +`trustLevel` (default 1/3) or more of last trusted validators signed correctly.
 		err := h1NextVals.VerifyCommitTrusting(chainID, h2.Commit.BlockID, h2.Height, h2.Commit, trustLevel)
 		if err != nil {
-			return err
+			return ErrNewValSetCantBeTrusted{err}
 		}
 	}
 
 	// Ensure that +2/3 of new validators signed correctly.
 	err := h2Vals.VerifyCommit(chainID, h2.Commit.BlockID, h2.Height, h2.Commit)
 	if err != nil {
-		return err
+		return ErrNewHeaderCantBeTrusted{err}
```

Comment:
Should we just return err here? Again assimilating with `ErrNewValSetCantBeTrusted` doesn't seem to be the right thing to do. IMO `ErrNewValSetCantBeTrusted` should only show on the skipping header case where we verify that > 1/3rd power of old validators have signed the new header, and should be the only case where bisection continues.

Reply:
> Again assimilating with ErrNewValSetCantBeTrusted doesn't seem to be the right thing to do.

NOTE these are 2 different errors - `ErrNewValSetCantBeTrusted` VS `ErrNewHeaderCantBeTrusted`

Reply:
Removed all new errors except ErrNewValSetCantBeTrusted

Reply:
> NOTE these are 2 different errors - ErrNewValSetCantBeTrusted VS ErrNewHeaderCantBeTrusted

Ah! Sorry, my bad.